src/openrct2/rct2/S6Importer.cpp

```diff
@@ -874,7 +874,11 @@ class S6Importer final : public IParkImporter
             bool invented = (_s6.researched_scenery_items[quadIndex] & ((uint32_t)1 << bitIndex));
 
             if (invented)
-                scenery_set_invented(sceneryEntryIndex);
+            {
+                ScenerySelection scenerySelection = { static_cast<uint8_t>((sceneryEntryIndex >> 8) & 0xFF),
+                                                      static_cast<uint16_t>(sceneryEntryIndex & 0xFF) };
```

Comment:
You [say](https://github.com/OpenRCT2/OpenRCT2/pull/10922#issuecomment-599203500) that 

> upper 8 bits being the scenery type (small, large, wall, path bit, banner)

but https://github.com/OpenRCT2/OpenRCT2/blob/7219c880e2db58a8a8d85cafafb5958a09512107/src/openrct2/management/Research.h#L67 and https://github.com/Gymnasiast/OpenRCT2/blob/567e70bfb67c06ed301e7869401e40d78fceef97/src/openrct2/rct2/RCT2.h#L35-L37 equals to `0x700`, so you end up setting `SceneryType` to 7.

Previously this code iterated over whole array, which did not even count a multiple of 5 (`small`, `large`, `wall`, `path bit`, `banner`), so there must be something off with your initial assumption
```suggestion
                                                      static_cast<uint16_t>(sceneryEntryIndex & 0xFF) };
                assert(scenerySelection.SceneryType < SCENERY_TYPE_COUNT);
```

Reply:
Thanks, writing beyond the confines of an array was indeed the cause. I have put the assert in `scenery_set_invented()` and used an `if` here, for reasons I explained on lines 881 and 882.